tempesta_fw/http.c

```diff
@@ -388,15 +389,17 @@ tfw_http_conn_msg_free(TfwHttpMsg *hm)
  * after the connection is dropped.
  */
 static inline void
-tfw_http_conn_cli_dropfree(TfwHttpMsg *hmreq)
+tfw_http_conn_cli_dropfree(TfwHttpMsg *hm)
 {
-	BUG_ON(!(TFW_CONN_TYPE(hmreq->conn) & Conn_Clnt));
+	BUG_ON(!(TFW_CONN_TYPE(hm->conn) & Conn_Clnt));
 
-	if (hmreq->flags & TFW_HTTP_CONN_CLOSE) {
-		if (ss_close(hmreq->conn->sk) == SS_OK)
-			tfw_sock_clnt_drop(hmreq->conn->sk);
+	if ((hm->flags & TFW_HTTP_CONN_CLOSE) ||
+	    (hm->version < TFW_HTTP_VER_11 && !(hm->flags & TFW_HTTP_CONN_KA)))
```

Comment:
It appears that this logic is already present in the correct place in a more recent version of `http.c`. Please see here: https://github.com/natsys/tempesta/blob/master/tempesta_fw/http.c#L779